src/config.rs

```diff
@@ -48,6 +52,44 @@ impl GitlabConfig {
     }
 }
 
+/// Fetch format for package metadata.
+#[derive(Default, Debug, Clone, Copy, PartialEq, Eq, Deserialize)]
+#[serde(rename_all = "lowercase")]
+pub enum MetadataFormat {
+    /// Plain json.
+    ///
+    /// Fetches `metadata.json` files.
+    #[default]
+    Json,
+    /// Json compressed with zstd.
+    ///
+    /// Fetches `metadata.json.zst` files.
+    #[serde(rename = "json.zst")]
+    JsonZst,
+}
+
+impl MetadataFormat {
+    #[must_use]
+    pub fn filename(self) -> &'static str {
+        match self {
+            Self::Json => "metadata.json",
+            Self::JsonZst => "metadata.json.zst",
+        }
+    }
+
+    pub async fn decode(self, res: reqwest::Response) -> anyhow::Result<cargo_metadata::Metadata> {
+        match self {
+            Self::Json => Ok(handle_error(res).await?.json().await?),
+            Self::JsonZst => {
+                let body = handle_error(res).await?.bytes().await?;
+                tokio::task::block_in_place(move || {
```

Comment:
Could we move ownership into here and use spawn_blocking please?

Reply:
Sure. I saw `block_in_place` as perhaps better as it seemed more likely there would be few other tasks ongoing while fetching metadata ... but it's more of a hunch so I'm not too fussed either way :smile: